webapp/channels/src/components/channel_bookmarks/utils.ts

```diff
@@ -124,8 +132,15 @@ export const useChannelBookmarks = (channelId: string) => {
         }
     }, [bookmarks]);
 
+    const reorder = async (id: string, prevOrder: number, nextOrder: number) => {
+        setTempOrder(insertWithoutDuplicates(order, id, nextOrder));
+        await dispatch(reorderBookmark(channelId, id, nextOrder));
+    };
+
     return {
         bookmarks,
-        order,
-    };
+        order: tempOrder ?? order,
```

Comment:
I chose to use local state inside the hook here mostly because `order` is a computed value as derived from the bookmarks redux state. But also I guess for a few other reasons:

- `setTempOrder(insertWithoutDuplicates(...))` and an Indiana Jones swap seemed simpler than reconstructing and patching each bookmark according to its ID and index position, of which `insertWithoutDuplicates` would be needed anyways.
- Keeps the temp value closer to the reorder operation and rendering flow, also keeping the redux state pure or reliably current. Also makes it possible to throw the temp value away, which IMO is easier than re-dispatching a tracked previous-version if something goes wrong.

Reply:
Thanks for the explanation. Those both make a ton of sense to me, and now I want to go back and do something similar for the channel order in the LHS since that's tremendously more complicated than this 😅